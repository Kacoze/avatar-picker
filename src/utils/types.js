import propTypes from 'prop-types';

const AvatarShape = propTypes.shape({
  src: propTypes.string.isRequired,
  label: propTypes.string.isRequired,
  id: propTypes.number.isRequired,
});

const AvatarListShape = propTypes.arrayOf(AvatarShape);

export { AvatarShape, AvatarListShape };
