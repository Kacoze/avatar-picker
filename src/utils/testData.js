const AvatarListExample = [
  { src: 'images/avatar/avatar1.png', label: 'Avatar 1', id: 1 },
  { src: 'images/avatar/avatar2.png', label: 'Avatar 2', id: 2 },
  { src: 'images/avatar/avatar3.png', label: 'Avatar 3', id: 3 },
  { src: 'images/avatar/avatar4.png', label: 'Avatar 4', id: 4 },
  { src: 'images/avatar/avatar5.png', label: 'Avatar 5', id: 5 },
  { src: 'images/avatar/avatar6.png', label: 'Avatar 6', id: 6 },
];
const AvatarExample = AvatarListExample[0];

export { AvatarListExample, AvatarExample };
