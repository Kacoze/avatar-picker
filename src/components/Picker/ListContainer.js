import React, { Component } from 'react';
import propTypes from 'prop-types';
import { AvatarListShape } from '../../utils/types';
import Avatar from './Avatar';
import styles from './style.scss';

class ListContainer extends Component {
  handleClickOutside = () => this.props.handleActiveClick();
  render() {
    const {
      list,
      activeAvatar,
      loading,
      handleAvatarChange,
      handleActiveClick,
    } = this.props;
    return (
      <div className={styles.Popover}>
        <h4 className={styles.PickerHeader}>Choose your avatar</h4>
        <div className={styles.AvatarList}>
          <Avatar
            image={list[activeAvatar - 1]}
            onAction={handleActiveClick}
          />
          {list.map(image => image.id !== list[activeAvatar - 1].id && (
            <Avatar
              key={image.id}
              image={image}
              onAction={handleAvatarChange}
              loading={loading === image.id}
            />
          ))}
        </div>
      </div>
    );
  }
}

ListContainer.propTypes = {
  list: AvatarListShape.isRequired,
  loading: propTypes.oneOfType([propTypes.bool, propTypes.number]).isRequired,
  activeAvatar: propTypes.number.isRequired,
  handleAvatarChange: propTypes.func.isRequired,
  handleActiveClick: propTypes.func.isRequired,
};

export default ListContainer;
