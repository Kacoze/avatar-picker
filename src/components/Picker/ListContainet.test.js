import React from 'react';
import renderer from 'react-test-renderer';
import { AvatarListExample } from '../../utils/testData';
import ListContainer from './ListContainer';


it('renders without crashing and match snapshoot', () => {
  const tree = renderer
    .create(<ListContainer
      list={AvatarListExample}
      activeAvatar={2}
      loading
      handleAvatarChange={() => {}}
      handleActiveClick={() => {}}
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
