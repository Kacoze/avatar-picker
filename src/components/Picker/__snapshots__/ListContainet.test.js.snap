// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing and match snapshoot 1`] = `
<div
  className="Popover"
>
  <h4
    className="PickerHeader"
  >
    Choose your avatar
  </h4>
  <div
    className="AvatarList"
  >
    <button
      className="avatarButton false"
      onClick={[Function]}
    >
      <img
        alt="Avatar 2"
        className="avatar"
        id={null}
        src="images/avatar/avatar2.png"
      />
    </button>
    <button
      className="avatarButton false"
      onClick={[Function]}
    >
      <img
        alt="Avatar 1"
        className="avatar"
        id={null}
        src="images/avatar/avatar1.png"
      />
    </button>
    <button
      className="avatarButton false"
      onClick={[Function]}
    >
      <img
        alt="Avatar 3"
        className="avatar"
        id={null}
        src="images/avatar/avatar3.png"
      />
    </button>
    <button
      className="avatarButton false"
      onClick={[Function]}
    >
      <img
        alt="Avatar 4"
        className="avatar"
        id={null}
        src="images/avatar/avatar4.png"
      />
    </button>
    <button
      className="avatarButton false"
      onClick={[Function]}
    >
      <img
        alt="Avatar 5"
        className="avatar"
        id={null}
        src="images/avatar/avatar5.png"
      />
    </button>
    <button
      className="avatarButton false"
      onClick={[Function]}
    >
      <img
        alt="Avatar 6"
        className="avatar"
        id={null}
        src="images/avatar/avatar6.png"
      />
    </button>
  </div>
</div>
`;
